src/NUnitFramework/framework/Internal/MessagePumpStrategy.cs

```diff
@@ -171,9 +173,7 @@ private sealed class SingleThreadedTestMessagePumpStrategy : MessagePumpStrategy
 
             public override void WaitForCompletion(AwaitAdapter awaiter)
             {
-                var context = SynchronizationContext.Current as SingleThreadedTestSynchronizationContext;
-
-                if (context == null)
+                if (!(SynchronizationContext.Current is SingleThreadedTestSynchronizationContext context))
                     throw new InvalidOperationException("This strategy must only be used from a SingleThreadedTestSynchronizationContext.");
```

Comment:
My preference
```suggestion
var context = SynchronizationContext.Current as SingleThreadedTestSynchronizationContext
    ?? throw new InvalidOperationException("This strategy must only be used from a SingleThreadedTestSynchronizationContext.");
```

Or, consider `is not` instead of `!(`